src/cpp/fastdds/xtypes/serializers/idl/dynamic_type_idl.cpp

```diff
@@ -0,0 +1,1086 @@
+    ret = dyn_type->get_descriptor(type_descriptor);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    bounds = type_descriptor->bound();
+
+    static const auto UNBOUNDED = static_cast<std::uint32_t>(LENGTH_UNLIMITED);
+
+    if (1 == bounds.size() && UNBOUNDED == bounds[0])
+    {
+        bounds.clear();
+    }
+
+    return ret;
+}
+
+//////////////////////////////
+// DYNAMIC TYPE TREE TO IDL //
+//////////////////////////////
+
+ReturnCode_t dyn_type_tree_to_idl(
+        const TreeNode<TreeNodeType>& root,
+        std::ostream& idl) noexcept
+{
+    ReturnCode_t ret = RETCODE_OK;
+
+    std::set<std::string> types_written;
+
+    // For every Node, check if it is of a "writable" type (i.e. struct, enum or union).
+    // If it is, check if it is not yet written
+    // If it is not, write it down
+    for (const auto& node : root.all_nodes())
+    {
+        if (types_written.find(node.info.type_kind_name) != types_written.end())
+        {
+            continue;
+        }
+
+        const auto kind = node.info.dynamic_type->get_kind();
+
+        switch (kind)
+        {
+            case TK_ALIAS:
+            {
+                ret = alias_to_idl(node, idl);
+                break;
+            }
+            case TK_BITMASK:
+            {
+                ret = bitmask_to_idl(node, idl);
+                break;
+            }
+            case TK_BITSET:
+            {
+                ret = bitset_to_idl(node, idl);
+                break;
+            }
+            case TK_ENUM:
+            {
+                ret = enum_to_idl(node, idl);
+                break;
+            }
+            case TK_STRUCTURE:
+            {
+                ret = struct_to_idl(node, idl);
+                break;
+            }
+            case TK_UNION:
+            {
+                ret = union_to_idl(node, idl);
+                break;
+            }
+            default:
+            {
+                continue;
+            }
+        }
+
+        if (RETCODE_OK != ret)
+        {
+            return ret;
+        }
+
+        idl << "\n";
+        types_written.insert(node.info.type_kind_name);
+    }
+
+    // Write struct parent node at last, after all its dependencies
+    ret = struct_to_idl(root, idl);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    return ret;
+}
+
+ReturnCode_t alias_to_idl(
+        const TreeNode<TreeNodeType>& node,
+        std::ostream& idl) noexcept
+{
+    assert(node.info.dynamic_type->get_kind() == TK_ALIAS);
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    ret = node.info.dynamic_type->get_descriptor(type_descriptor);
+
+    if (RETCODE_OK != ret)
+    {
+        return ret;
+    }
+
+    idl << "typedef ";
+
+    // Find the base type of the alias
+    ret = type_kind_to_idl(type_descriptor->base_type(), idl);
```

Comment:
Be careful here, we must make sure typedef dependencies are written in the right order to produce a valid IDL.